tests/post_training/pipelines/image_classification_torchvision.py

```diff
@@ -0,0 +1,169 @@
+
+        if self.backend == BackendType.FX_TORCH:
+            with torch.no_grad():
+                with disable_patching():
+                    self.model = capture_pre_autograd_graph(model, (torch.ones(self.input_size),))
+
+        elif self.backend in PT_BACKENDS:
+            self.model = model
+
+        elif self.backend in [BackendType.OV, BackendType.FP32]:
+            with torch.no_grad():
+                self.model = ov.convert_model(model, example_input=self.dummy_tensor, input=self.input_size)
+            self.input_name = list(inp.get_any_name() for inp in self.model.inputs)[0]
+
+        self._dump_model_fp32()
+
+        # Set device after dump fp32 model
+        if self.backend == BackendType.CUDA_TORCH:
+            self.model.cuda()
+            self.dummy_tensor = self.dummy_tensor.cuda()
+
+    def _dump_model_fp32(self) -> None:
+        """Dump IRs of fp32 models, to help debugging."""
+        if self.backend in PT_BACKENDS:
+            with disable_patching():
+                ov_model = ov.convert_model(
+                    torch.export.export(self.model, args=(self.dummy_tensor,)),
+                    example_input=self.dummy_tensor,
+                    input=self.input_size,
+                )
+            ov.serialize(ov_model, self.fp32_model_dir / "model_fp32.xml")
+
+        if self.backend == BackendType.FX_TORCH:
+            exported_model = torch.export.export(self.model, (self.dummy_tensor,))
+            ov_model = ov.convert_model(exported_model, example_input=self.dummy_tensor, input=self.input_size)
+            ov.serialize(ov_model, self.fp32_model_dir / "fx_model_fp32.xml")
+
+        if self.backend in [BackendType.FP32, BackendType.OV]:
+            ov.serialize(self.model, self.fp32_model_dir / "model_fp32.xml")
+
+    def prepare_preprocessor(self) -> None:
+        self.transform = self.model_weights.transforms()
+
+    def get_transform_calibration_fn(self):
+        if self.backend in [BackendType.FX_TORCH] + PT_BACKENDS:
+            device = torch.device("cuda" if self.backend == BackendType.CUDA_TORCH else "cpu")
+
+            def transform_fn(data_item):
+                images, _ = data_item
+                return images.to(device)
+
+        else:
+
+            def transform_fn(data_item):
+                images, _ = data_item
+                return {self.input_name: np.array(images, dtype=np.float32)}
+
+        return transform_fn
+
+    def prepare_calibration_dataset(self):
+        dataset = datasets.ImageFolder(root=self.data_dir / "imagenet" / "val", transform=self.transform)
+        loader = torch.utils.data.DataLoader(dataset, batch_size=self.batch_size, num_workers=2, shuffle=False)
+
+        self.calibration_dataset = nncf.Dataset(loader, self.get_transform_calibration_fn())
+
+    def _validate(self):
+        val_dataset = datasets.ImageFolder(root=self.data_dir / "imagenet" / "val", transform=self.transform)
```

Comment:
Looks like it can be reused from image_classification_timm.py

Reply:
Done